hubstorage/resourcetype.py

```diff
@@ -40,7 +44,7 @@ def apiget(self, _path=None, **kwargs):
         return self.apirequest(_path, method='GET', **kwargs)
```

Comment:
`DELETE` is a special case, it should use a specific & more complex retrier that turn 404 into "everything went fine" (whether to accept 404 on the first attempt is worth a discussion since it would require changing/not using the retrying library).

But the current use of `DELETE`, from `apidelete` never return 404 (except for a very specific case that looks like a bug, see SC-338). That's why I handled it as a special case of idempotence. (there's a `test_delete_on_hubstorage_api_does_not_404` to verify that).

I do see the thinking behind "let _iter_lines just forward the request parameters to the client.request". My only last, tiny, concern, if we think like that: we should set the request timeout in the client.request, not in _iter_lines it's not the role of the resource to setup this value.

 I'll do as you say, we'll come back later to `iter_json` with more tests.